src/description/description/common.xacro

```diff
@@ -0,0 +1,232 @@
+                        <box size="${hand_length} ${hand_width} ${hand_height}"/>
+                    </geometry>
+                    <origin xyz="0 0.5 0"/>
+                </collision>
+                <inertial>
+                    <mass value="0.2"/>
+                    <origin xyz="0 0 0"/>
+                    <inertia ixx="0.0005" ixy="0.0" ixz="0.0" iyy="0.0005" iyz="0.0" izz="0.0005"/>
+                </inertial>
+                <gazebo>
+                    <material>Gazebo/Orange</material>
+                </gazebo>
+            </link>
+
+            <joint name="${name}_joint" type="fixed">
+                <parent link="${parent_link}"/>
+                <child link="${name}"/>
+                <origin xyz="0 0.5 0"/>
+            </joint>
+        </xacro:macro>
+
+    <!-- Solenoid Macro -->
+        <xacro:macro name="solenoid" params="name parent_link">
+            <link name="${name}">
+                <visual>
+                    <geometry>
+                        <cylinder radius="0.01" length="0.025"/>
+                    </geometry>
+                    <material name="green"/>
+                </visual>
+                <inertial>
+                    <mass value="0.05"/>
+                    <origin xyz="0 0 0"/>
+                    <inertia ixx="0.00005" ixy="0.0" ixz="0.0" iyy="0.00005" iyz="0.0" izz="0.00005"/>
+                </inertial>
+                <gazebo>
+                    <material>Gazebo/Green</material>
+                </gazebo>
+            </link>
+            <joint name="${name}_joint" type="revolute">
+                <parent link="${parent_link}"/>
+                <child link="${name}"/>
+                <origin xyz="0 0 0"/>
+                <axis xyz="1 0 0"/>
+                <limit lower="-0.2" upper="0.2" effort="1" velocity="1.0"/>
+            </joint>
+        </xacro:macro>
+
+    <!-- Servo Macro -->
+        <xacro:macro name="servo" params="name parent_link x_origin">
+            <link name="${name}">
+                <visual>
+                    <geometry>
+                        <box size="0.02 0.02 0.02"/>
+                    </geometry>
+                    <material name="yellow"/>
+                </visual>
+                <inertial>
+                    <mass value="0.05"/>
+                    <origin xyz="0 0 0"/>
+                    <inertia ixx="0.00002" ixy="0.0" ixz="0.0" iyy="0.00002" iyz="0.0" izz="0.00002"/>
+                </inertial>
+                <gazebo>
+                    <material>Gazebo/Yellow</material>
+                </gazebo>
+            </link>
+            <joint name="${name}_joint" type="revolute">
+                <parent link="${parent_link}"/>
+                <child link="${name}"/>
+                <origin xyz="${x_origin} -${hand_width/2} -0.01"/>
+                <axis xyz="0 0 1"/>
+                <limit lower="-0.2" upper="0.2" effort="1" velocity="1.0"/>
+            </joint>
+        </xacro:macro>
+    
```

Comment:
I don't think the common/core URDF definitions should be coupled to the hardware, and thus recommend renaming the servo and solenoid joints

Reply:
I am not sure if I understand what you mean.